osbrain/agent.py

```diff
@@ -1743,41 +1762,44 @@ def has_socket(self, alias):
         """
         return alias in self._socket
 
-    def close(self, alias):
+    def _delete_socket_entries(self, entries):
+        for entry in entries:
+            del self._socket[entry]
+
+    def close(self, alias, linger=None):
         """
         Close a socket given its alias and clear its entry from the
         `Agent.socket` dictionary.
         """
+        if linger is None:
```

Comment:
As mentioned in my other comment it is to set linger to zero when killing (i.e.: killing should be fast).

This would definitely need more tests. I do not think we need to create a new issue, as there is #255 already.

Anyway I think you are right. I will either add some tests or undo that linger change (and remove the `_die()` method, which would make little sense then.